astrapy/utils.py

```diff
@@ -182,23 +185,100 @@ def convert_vector_to_floats(vector: Iterable[Any]) -> List[float]:
     return [float(value) for value in vector]
 
 
-def preprocess_insert(document: Dict[str, Any]) -> Dict[str, Any]:
+def is_list_of_floats(vector: Iterable[Any]) -> bool:
     """
-    Perform preprocessing operations before an insertion
+    Safely determine if it's a list of floats.
+    Assumption: if list, and first item is float, then all items are.
+    """
+    if isinstance(vector, list):
+        if len(vector):
+            return isinstance(vector[0], float) or isinstance(vector[0], int)
+        else:
+            return True
+    else:
+        return False
```

Comment:
Possible one-liner (not always a good idea but I think here it's readable)
```suggestion
    return isinstance(vector, list) and (len(vector) == 0 or isinstance(vector[0], float) or isinstance(vector[0], int))
```